File: math/13_RomanToInteger/Solution.cpp
class Solution {
public:
    int romanToInt(string s) {
        
        map<char, int> roman = {
            { 'I', 1 }, { 'V', 5 }, { 'X', 10 }, { 'L', 50 },
            { 'C', 100 }, { 'D', 500 }, { 'M', 1000 }
        };
        int i = 0;
        int num = 0;
        char prev;
         
        for(char& l : s) {
            
            if (
                (prev == 'I' and (l == 'V' or l == 'X')) or
                (prev == 'X' and (l == 'L' or l == 'C')) or
                (prev == 'C' and (l == 'D' or l == 'M'))
               ) {
                
                 num = num + (roman[l] - (roman[prev] * 2));
                
            } else {
                
                 num = num + roman[l];
                
            }
                    
            prev = l;
                    
        }
        
        return num;
    }
};

File: math/13_RomanToInteger/Solution.py
class Solution:
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        roman = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}

        num = 0
        prev = ""
        for l in s:

            if (prev == "I" and l in ["V", "X"]) or \
                (prev == "X" and l in ["L", "C"]) or \
                (prev == "C" and l in ["D", "M"]):
                    
                num = num + (roman[l] - (roman[prev] * 2))

            else:
                num = num + roman[l]
            
            prev = l

        return num
